presto-docs/src/main/sphinx/admin/session-property-managers.rst

```diff
@@ -0,0 +1,90 @@
+=========================
+Session Property Managers
+=========================
+
+Administrators can add session properties to control the behavior for subsets of their workload.
+These properties are defaults and can be overridden by users (if authorized to do so). Session
+properties can be used to control resource usage, enable or disable features, and change query
+characteristics. Session property managers are pluggable.
+
+Add an ``etc/session-property-config.properties`` file with the following contents to enable
+the built-in manager that reads a JSON config file:
+
+.. code-block:: none
+
+    session-property-config.configuration-manager=file
+    session-property-manager.config-file=etc/session-property-config.json
+
+Change the value of ``session-property-manager.config-file`` to point to a JSON config file,
+which can be an absolute path, or a path relative to the Presto data directory.
+
+This configuration file consists of a list of match rules, each of which specify a list of
+conditions that the query must meet, and a list of session properties that should be applied
+by default. All matching rules contribute to constructing a list of session properties. Rules
+are applied in the order they are specified. Rules specified later in the file override values
+for properties that have been previously encountered.
+
+Match Rules
+-----------
+
+* ``user`` (optional): regex to match against user name. Defaults to ``.*``
+
+* ``source`` (optional): regex to match against source string. Defaults to ``.*``
+
+* ``queryType`` (optional): string to match against the type of the query submitted. The query type can be:
+    * ``DATA_DEFINITION``: Queries that alter/create/drop the metadata of schemas/tables/views, and that manage
+      prepared statements, privileges, sessions, and transactions.
+    * ``DELETE``: ``DELETE`` queries.
+    * ``DESCRIBE``: ``DESCRIBE``, ``DESCRIBE INPUT``, ``DESCRIBE OUTPUT``, and ``SHOW`` queries.
+    * ``EXPLAIN``: ``EXPLAIN`` queries.
+    * ``INSERT``: ``INSERT`` and ``CREATE TABLE AS SELECT`` queries.
+    * ``SELECT``: ``SELECT`` queries.
+
+* ``clientTags`` (optional): list of tags. To match, this must be a subset of the client-provided tags associated
```

Comment:
Maybe change this to "To match, every tag in this list must be in the list of client-provided tags associated with the query."